test/factories/editions.rb

```diff
@@ -161,6 +161,12 @@
         edition.unpublishing = build(:withdrawn_unpublishing, edition: edition)
       end
     end
+
+    trait(:locked) do
```

Comment:
I'm a bit unsure about this being a trait for edition because it's the document that's locked, not the edition. @brucebolt do you have any thoughts on this?

Reply:
I agree with @emmabeynon.  Adding a locked trait to the edition reads like you're created a locked edition, with is not the case.  I think I'd rather the edition were created, then the document locked separately after that.

Reply:
This was down to @kevindew based on the fact that creating the edition is a two step problem and we really need to save the document lock after the edition is saved, else we cannot save said edition.

Reply:
> I agree with @emmabeynon. Adding a locked trait to the edition reads like you're created a locked edition, with is not the case. I think I'd rather the edition were created, then the document locked separately after that.

Yeah that was my original intent.

Reply:
Yup, this was my contribution. Tada! 😁 

We could rename this trait to be `create(:edition, :with_locked_document)` which is consistent with: https://github.com/alphagov/whitehall/blob/master/test/factories/editions.rb#L130-L132

Though given we have a method of `edition.locked?` that is already implying we treat the edition as locked rather than the document. So I see `create(:edition, :locked)` to be pretty much analogous to `edition.locked?`.

I don't mind too much though if `:with_locked_document` or something different satisfies concerns. I do think think a trait is preferable to a multi step process in each test though.

Reply:
Incidentally I tried to get rid of the `with_document` trait as that seemed rather unnecessary since documents are created automatically: https://github.com/alphagov/whitehall/commit/972c949742864c817277bd638d43dd47609172ef - however the amount of test failures that would need to be resolved discouraged me from pursuing it further.

Reply:
I think `:with_locked_document` would be clearer to me.  I understand what you mean about having `edition.locked?` already, though it was really to avoid having to repeat `edition.document.locked?` all the time.

Reply:
Will be in next update.